### PR TITLE
Add parameter -s

### LINKED ISSUE #4: Parameter -s

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #4

### CODE REVIEW

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Rebase. This time with -i – in the editor remove all the commits having nothing to do with this PR.

File: main.go [hunk 1]
```diff
@@ -81,6 +82,13 @@ func main() {
 		lineLength := len(data)
 		var output [][]byte
 
+		if lineLength < 2 && *onlyDelimitParameter {
```
Comment (Al2Klimov) [inline main.go]: Please explain this condition.

Comment (justinmueller05082000) [reply main.go]: well the second one is not hard to understand and the first one is there to check if there are several values in one line separated by a delimiter

Comment (Al2Klimov) [reply main.go]: Which several values? Shouldn't you check for the delimiter?

https://github.com/Al2Klimov/trainee-go-cut/pull/13/commits/a0d402f8bd778160a43c6de8a6a69cea4f523156#diff-7ddfb3e035b42cd70649cc33393fe32cR19

File: main.go [hunk 2]
```diff
@@ -69,13 +70,23 @@ func main() {
...
 		}
 
 		contentLineSplit := bytes.Split(content, []byte{'\n'})
+
+		if *onlyDelimitParameter {
+			for y := range contentLineSplit[0] {
+				if contentLineSplit[0][y] == delimiter[0] {
+					onlyDelimitCheck = true
+				}
+			}
```
Comment (Al2Klimov) [inline main.go]: Isn't there a smarter way to check whether a contains b?

File: main.go [hunk 3]
```diff
@@ -92,7 +103,9 @@ func main() {
 
 		contentLineSplit[0] = bytes.Join(output, delimiter)
 
-		_, _ = os.Stdout.Write(bytes.Join(contentLineSplit, []byte{'\n'}))
+		if onlyDelimitCheck {
+			_, _ = os.Stdout.Write(bytes.Join(contentLineSplit, []byte{'\n'}))
+		}
```
Comment (Al2Klimov) [inline main.go]: Shouldn't this if cover more code for performance reasons?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (justinmueller05082000) [reply main.go]: bytes.Contains?

Comment (Al2Klimov) [reply main.go]: Exactly.

File: main.go [hunk 4]
```diff
@@ -69,33 +70,43 @@ func main() {
...
 
-		lineLength := len(data)
-		var output [][]byte
+		if *onlyDelimitParameter {
+			if bytes.Contains(contentLineSplit[0], delimiter) {
+				onlyDelimitCheck = true
+			}
+		}
+
+		if onlyDelimitCheck {
```
Comment (Al2Klimov) [inline main.go]: Any way to shorten this to a one-liner?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (justinmueller05082000) [reply main.go]: if bytes.Contains(contentLineSplit[0], delimiter) && *onlyDelimitParameter {
}

Comment (Al2Klimov) [reply main.go]: Exactly.

Comment (justinmueller05082000) [reply main.go]: but if i want to just use one parameter later on like e.g -f, -c -> *onlyDelimitParameter will be always false and so on the program wont work

Comment (Al2Klimov) [reply main.go]: But then your condition was already wrong and this has nothing to do with the shortening I requested.

Comment (justinmueller05082000) [reply main.go]: u may be right with that

File: main.go [hunk 5]
```diff
@@ -76,26 +77,29 @@ func main() {
 		}
 
 		contentLineSplit := bytes.Split(content, []byte{'\n'})
-		data = bytes.Split(contentLineSplit[0], delimiter)
 
-		lineLength := len(data)
-		var output [][]byte
+		if (bytes.Contains(contentLineSplit[0], delimiter) && *onlyDelimitParameter) || !*onlyDelimitParameter {
```
Comment (Al2Klimov) [inline main.go]: Any way to shorten this condition?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 5 above)
Comment (justinmueller05082000) [reply main.go]: i dont tihnk so

Comment (justinmueller05082000) [reply main.go]: remove the brackets?

Comment (Al2Klimov) [reply main.go]: Swap the two expressions of the ||. Maybe then you'll see the way I mean.

Comment (justinmueller05082000) [reply main.go]: still dont get it

Comment (Al2Klimov) [reply main.go]: After the swapping the left and right side of the || remove the right side of the &&. That's what I mean.

Comment (justinmueller05082000) [reply main.go]: if (bytes.Contains(contentLineSplit[0], delimiter) && || *onlyDelimitParameter {
?????

Comment (Al2Klimov) [reply main.go]: No, `if !*onlyDelimitParameter || bytes.Contains(contentLineSplit[0], delimiter) {`.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Disallow -s with -c.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD feature/add-copying-file-9)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/add-parameter-s-4`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toCut.txt:**
```
abc:defgh
deutschland:griechenland
hundkatze-maus
linux:windows:ubuntu
```

**Go command:**

**Terminal:** `go run . -s -d : -f 1 < toCut.txt`

**Output:**
```
abc
deutschland
linux
```

**Cut command:**

**Terminal:** `cut -s -d : -f 1 < toCut.txt`

**Output:**
```
abc
deutschland
linux
```

**Conclusion:**
Both results are equal.

**Review by Al2Klimov — APPROVED** [review]
